man/fi_profile.3.md

```diff
@@ -0,0 +1,260 @@
+
+*varlist*
+: An array of profiling variables.
+
+*eventlist*
+: An array of profiling events.
+
+*count*
+: Number of variables or events in the array.
+
+*var_id*
+: An unqie ID for a variable.
+
+*event*
+: A unqie ID for an event. 
+
+*callback*
+: Callback function registered for an event.
+
+*param*
+: The parameter of the callback function.
+
+*size*
+: The amount of the data the provider is provided in the "param" 
+when the callback is called.
+
+*data*
+: An uint64_t value of a variable.
+
+# DESCRIPTION
+
+Fabric profiling interface allows users to obtain some of configuration or
+runtime properties from the providers in the form of "variables". Providers 
+define variables and make them readable to users through the profiling 
+interface. Providers define events and use them to report an operation 
+occured or a state changed where one or more variables may change their 
+values. Users can register callback functions to one or more events, and 
+receive notifications from providers once an event occurs.
+
+In order to support the profiling interface, an endpoing or a domain in a 
+provider needs to maintain a fid_profile object, and implement the profiling 
+operations there. To starts a profiling,  an open call must be made on the 
+target endpoint or domain object to obtain the fid_profile object.  If the 
+target object does not support profiling, the open call will fail with error 
+code -FI_ENOSYS. Once a fid_profile object is obtained successfully, it can be 
+used to query available variables or events, and read variables' values.
+
+# VARIABLES
+
+The profiling variables are defined in providers, and become accessable 
+(readable) externally through the profiling interface.  A variable can be 
+a configuration setting, a runtime state or a counter in an endpoint or a 
+domain in a provider. Some variables are common and available in most 
+providers, while others are provider-specific, and are only available in 
+one provider. Users can query available variables and their values using 
+the profiling interface.
+
+FI_VAR_UNEXP_MSG_CNT is a common variable and it counts the number of  
+unexpected message recevied and unmatched. The variable has a unint64_t 
+data value.
+
+A variable is specified using struct fi_profile_desc and is described below.
+
+# EVENTS
+
+The profiling event is defined by the providers. It is used to notify users 
+that an operation has occured or a state has been moved, which may change 
+the values of some variables. Event is a way for providers to inform changes 
+on the variables. Similar to the variables, some events are pre-defined and 
+shared by many providers, while others are provider-specific events and 
+defined by each provider. Users can register a callback function on an event 
+to get notified when the event occurs.
+
+FI_EVENT_UNEXP_MSG_RECVD and FI_EVENT_UNEXP_MSG_MATCHED are common events 
+which imply there are changes on variable FI_VAR_UNEXP_MSG_CNT.
```

Comment:
Rework similar to above to give these their own sections:
```
## FI_EVENT_UNEXP_MSG_RECVD (param = ?)

This event indicates that an unexpected message was received and has been
queued by the provider.

## FI_EVENT_UNEXP_MSG_MATCHED (param = ?)

This event indicates that an unexpected messages that was previously queued
by the provider has been matched.
 ```
We need to decide what information is passed to both of the above events.  This might
be a completion structure (e.g. fi_cq_tagged_entry) that contains the src_addr + tag + size, etc.